### PR TITLE
Filter

### PR DESCRIPTION
**Pull request by evanyang1**
A series of pushes that will enable search of conversations and contacts. 

Done: 
Backend functionality that searches via Mongoose the database using basic regex. Also filters out only conversations that you are in (Person #1 isn't interested in a conversation between #2 and #3).

To do:
Implement similar functionality for contacts, and invites (inbound and outbound). Adapt the frontend to utilize these backend additions. 

Remarks:
It's difficult to search based on other members and NOT your own name. That would be on the backburner but if time permits, that task can be tackled.

### CODE REVIEW

File: routes/conversation.js [hunk 1]
```diff
@@ -83,6 +82,26 @@ router.put('/read/:id', auth, async (req, res) => {
   }
 });
 
-router.delete('/delete/:id', (req, res) => {});
+// get all conversations. Given a search parameter (optional)
+router.get('/search/:searchterm', auth, async (req, res) => {
```
Comment (bonnieli) [inline routes/conversation.js]: we don't want to introduce a new route for this. instead reuse /users/conversation route

Comment (bonnieli) [inline routes/conversation.js]:
```suggestion
router.get('/search', auth, async (req, res) => {
  const search = req.query.search
```

Comment (bonnieli) [reply routes/conversation.js]: https://github.com/hatchways/team-cocoa-puffs/blob/dev/routes/user.js#L172

File: routes/conversation.js [hunk 2]
```diff
@@ -83,6 +82,26 @@ router.put('/read/:id', auth, async (req, res) => {
...
+  const search = req.params.searchterm;
+  const user = req.user.id;
+  try {
+
+    const regex = new RegExp(search, 'i');
+    let conversations = await Conversation.find({
+      'members.name': { $regex: regex },
+      'members._id': user,
+    });
+    if (!conversations) return res.status(404).send('No conversations found');
```
Comment (bonnieli) [inline routes/conversation.js]: Instead of an error code, maybe just return an empty array signifying no conversations were found

**Comment by evanyang1** [comment]
conversation search done. invites, we'll leave that for the backburner

**Pull request merged**